src/CMakeLists.txt

```diff
@@ -88,6 +88,7 @@ target_link_libraries(
     $<$<BOOL:${DLAF_WITH_HIP}>:roc::rocprim>
     $<$<LINK_LANGUAGE:HIP>:roc::rocthrust> # hip adds offload-arch globally for the CXX language in hip-config
     $<$<BOOL:${DLAF_WITH_GPU}>:whip::whip>
+    $<$<BOOL:${DLAF_WITH_HDF5}>:hdf5_cpp-shared>
```

Comment:
Is `-shared` a requirement or a preference? What happens if `hdf5` is built with `~shared`?

Reply:
Actually, if we want to go for using `hdf5` shared libraries we should add it as requirement in `dla-future` spack package, because currently it is not enforced (by default in spack it is `hdf5+shared`).

About preference, not really, I just thought that not having them statically linked might have been a good idea. Do you have any suggestion about it?

Reply:
Not really. I'd expect both to work the same, but supporting both may need some additional CMake logic (to detect if the static or shared or both libraries are available)? How about we enforce `depends_on("hdf +shared", when="...")` for now with a comment since I assume the HDF5 option is not going to upstream spack (yet...)?

Reply:
Fully agree. See https://github.com/eth-cscs/DLA-Future/pull/880/commits/b57d53a1a5f726e1d906af0a3b63af8328dd15c1